src/Marten.AsyncDaemon.Testing/AsyncDaemonHealthCheckExtensionsTests.cs

```diff
@@ -0,0 +1,155 @@
+    {
+        StoreOptions(x =>
+        {
+            x.Projections.Add(new FakeSingleStream1Projection(), ProjectionLifecycle.Async);
+        });
+        var agent = await StartDaemon();
+        using var session = theStore.LightweightSession();
+        session.Events.Append(Guid.NewGuid(), new FakeIrrellevantEvent());
+        await session.SaveChangesAsync();
+        await agent.Tracker.WaitForHighWaterMark(1);
+        var healthCheck = new AsyncDaemonHealthCheck(theStore, new(100));
+
+        var result = await healthCheck.CheckHealthAsync(new());
+
+        result.Status.ShouldBe(HealthStatus.Healthy);
+    }
+
+    [Fact]
+    public async Task should_be_unhealty_with_no_projection_lag_allowed()
+    {
+        StoreOptions(x =>
+        {
+            x.Projections.Add(new FakeSingleStream2Projection(), ProjectionLifecycle.Async);
+        });
+        var agent = await StartDaemon();
+        using var session = theStore.LightweightSession();
+        var stream = Guid.NewGuid();
+        var eventCount = 100;
+        for (var i = 0; i < eventCount; i++)
+            session.Events.Append(stream, new FakeEvent());
+        await session.SaveChangesAsync();
+        await agent.Tracker.WaitForHighWaterMark(eventCount);
+        await agent.Tracker.WaitForShardState(new ShardState("FakeStream2:All", eventCount), 15.Seconds());
+        var healthCheck = new AsyncDaemonHealthCheck(theStore, new(0));
+
+        var result = await healthCheck.CheckHealthAsync(new());
+
+        result.Status.ShouldBe(HealthStatus.Unhealthy);
+    }
+
+
+    [Fact]
+    public async Task should_be_healty_with_all_projections_caught_up()
+    {
+
+        StoreOptions(x =>
+            {
+                x.Projections.Add(new FakeSingleStream3Projection(), ProjectionLifecycle.Async);
+                x.Projections.Add(new FakeSingleStream4Projection(), ProjectionLifecycle.Async);
+            });
+        var agent = await StartDaemon();
+        using var session = theStore.LightweightSession();
+        var stream1 = Guid.NewGuid();
+        var stream2 = Guid.NewGuid();
+        var eventCount = 100;
+        for (var i = 0; i < eventCount; i++)
+        {
+            session.Events.Append(stream1, new FakeEvent());
+            session.Events.Append(stream2, new FakeEvent());
+        }
+        await session.SaveChangesAsync();
+        await agent.Tracker.WaitForShardState(new ShardState("FakeStream3:All", eventCount), 15.Seconds());
+        await agent.Tracker.WaitForShardState(new ShardState("FakeStream4:All", eventCount), 15.Seconds());
+        await agent.Tracker.WaitForHighWaterMark(eventCount);
+        var healthCheck = new AsyncDaemonHealthCheck(theStore, new(1));
+
+        var result = await healthCheck.CheckHealthAsync(new());
+
+        result.Status.ShouldBe(HealthStatus.Healthy);
+    }
+
+    [Fact]
+    public async Task should_be_healty_with_one_projections_lagging()
```

Comment:
After looking at the assert statement, I believe this should be unhealthy, not healthy.

Reply:
You're right - typo. Good catch